src/xdp-utils.c

```diff
@@ -654,29 +721,16 @@ xdp_validate_serialized_icon (GVariant  *v,
 
   args[0] = icon_validator;
   args[1] = "--sandbox";
-  args[2] = name;
-  args[3] = NULL;
+  args[2] = NULL;
```

Comment:
Can we make the validator take either a file path or a fd argument and then mmap either of them? That way using it on the command line still works which will be helpful if this something is going wrong.

Reply:
I was thinking about doing that but it's possible to  just do `./_build/src/xdg-desktop-portal-validate-icon 3< image.png` on the terminal so since the validator is only used by us there isn't much need to have a nicer command line interface.

Reply:
Yes, but hardcoding some fd number is just bad design and having a filename argument takes a few lines of code.

Reply:
I added two options `--path` and `--fd` to both validators.